pkg/image/apis/image/helper.go

```diff
@@ -523,6 +539,44 @@ func ImageWithMetadata(image *Image) error {
 	return nil
 }
 
+// FetchImageSignature attemps to fetch the signature blob for the provided
+// image and mutates the image object with the signature content if the registry
+// has defined a known signature store.
+func FetchImageSignature(registry string, image *Image) error {
+	for name, sigStore := range knownSignatureStores {
+		if registry != name {
+			continue
+		}
```

Comment:
Why is this iterating through the map instead of a single `knownSignatureStores[registry]` lookup?